include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,458 @@
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary % operator to compute remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator %(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary %= operator to compute/assign remainder of an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left modulo \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator %=(Angle& left, Angle right);
+
+inline namespace literals
+{
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in degrees
+///
+/// \param angle Angle in degrees
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _deg(unsigned long long angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(long double angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief User defined literal for angles in radians
+///
+/// \param angle Angle in radians
+///
+/// \return \a Angle
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator "" _rad(unsigned long long angle);
+
+} // namespace literals
+
+#include <SFML/System/Angle.inl>
+
+} // namespace sf
+
+
+#endif // SFML_ANGLE_HPP
+
+
+////////////////////////////////////////////////////////////
+/// \class sf::Angle
+/// \ingroup system
+///
+/// sf::Angle encapsulates an angle value in a flexible way.
+/// It allows to define an angle value either as a number of
+/// degrees or radians. It also works the other way round:
+/// you can read an angle value as either a number of
+/// degrees or radians.
+///
+/// By using such a flexible interface, the API doesn't
+/// impose any fixed type or resolution for angle values,
+/// and let the user choose their own favorite representation.
```

Comment:
"resolution" is not really true, we do limit the implementation to 32-bit floats.

Maybe we should mention "...the API doesn't impose any unit for angle values..."?

Reply:
That's a better way to put it. For it's worth, `sf::Time` also uses similar language which also doesn't correctly represent what the code is doing. I doubt anybody is excited to comb through documentation comments to find these kinds of inaccuracies but it's probably another worthwhile task to do in the 3.x release series even if it doesn't land for 3.0.